File: RegexFileSorter/SortedFolder.vb
Public Class SortedFolder

	Public Sub New()
		Name = "No Name"
		Path = "None"
		Status = "None"
	End Sub

	Public Sub New(name As String)
		Me.Name = name
		Me.IsValid = False
	End Sub

	Public ReadOnly Property Files As New List(Of SortedFile)
	Public Property IsValid As Boolean
	Public Property Name As String
	Public Property Path As String
	Public Property Status As String

	Public Structure SortedFile

		Public Sub New(fileName As String, inPath As String, outPath As String)
			Me.FileName = fileName
			Me.InPath = inPath
			Me.OutPath = outPath
		End Sub

		Public Property FileName As String
		Public Property InPath As String
		Public Property OutPath As String
	End Structure

End Class

File: RegexFileSorterSharp/SortedFolder.cs
using System.Collections.Generic;

namespace RegexFileSorter
{
    internal class SortedFolder
    {
        public SortedFolder()
        {
            Name = "No Name";
            Path = "None";
            Status = "None";
        }

        public SortedFolder(string name)
        {
            Name = name;
            IsValid = false;
        }

        public List<SortedFile> Files { get; }
        public bool IsValid { get; set; }
        public string Name { get; set; }
        public string Path { get; set; }
        public string Status { get; set; }
        public record SortedFile(string Filename, string InPath, string OutPath);
    }
}